src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+    testObjectCreateSequence(FeatureBitset features)
+    {
+        testcase("object create w/ sequence");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const gw = Account("gw");
+        auto const USD = gw["USD"];
+
+        env.fund(XRP(1000), alice, bob, gw);
+        env.close();
+
+        env.trust(USD(1000), alice, bob);
+        env(pay(gw, alice, USD(100)));
+        env(pay(gw, bob, USD(100)));
+        env.close();
+
+        auto const aliceSeq = env.seq(alice);
+        auto const bobSeq = env.seq(bob);
+        auto const preAlice = env.balance(alice);
+        auto const preBob = env.balance(bob);
+        auto const preAliceUSD = env.balance(alice, USD.issue());
+        auto const preBobUSD = env.balance(bob, USD.issue());
+
+        auto const batchFee = batch::calcBatchFee(env, 1, 2);
+        uint256 const chkId{getCheckIndex(bob, env.seq(bob))};
+        env(batch::outer(alice, aliceSeq, batchFee, tfAllOrNothing),
+            batch::inner(check::create(bob, alice, USD(10)), bobSeq),
+            batch::inner(check::cash(alice, chkId, USD(10)), aliceSeq + 1),
+            batch::sig(bob));
+        auto const txIDs = env.tx()->getBatchTransactionIDs();
+        TxID const parentBatchId = env.tx()->getTransactionID();
+        std::vector<TestBatchData> testCases = {
+            {"tesSUCCESS", to_string(txIDs[0])},
+            {"tesSUCCESS", to_string(txIDs[1])},
+        };
+        env.close();
+        validateBatch(env, parentBatchId, testCases);
+
+        // Alice consumes sequences (# of txns)
+        BEAST_EXPECT(env.seq(alice) == aliceSeq + 2);
+
+        // Alice consumes sequences (# of txns)
+        BEAST_EXPECT(env.seq(bob) == bobSeq + 1);
+
+        // Alice pays Fee; Bob XRP Unchanged
+        BEAST_EXPECT(env.balance(alice) == preAlice - batchFee);
+        BEAST_EXPECT(env.balance(bob) == preBob);
+
+        // Alice pays USD & Bob receives USD
+        BEAST_EXPECT(env.balance(alice, USD.issue()) == preAliceUSD + USD(10));
+        BEAST_EXPECT(env.balance(bob, USD.issue()) == preBobUSD - USD(10));
+    }
+
+    void
+    testObjectCreateTicket(FeatureBitset features)
+    {
+        testcase("object create w/ ticket");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const gw = Account("gw");
+        auto const USD = gw["USD"];
+
+        env.fund(XRP(1000), alice, bob, gw);
+        env.close();
+
+        env.trust(USD(1000), alice, bob);
+        env(pay(gw, alice, USD(100)));
+        env(pay(gw, bob, USD(100)));
+        env.close();
+
+        std::uint32_t bobTicketSeq{env.seq(bob) + 1};
+        env(ticket::create(bob, 10));
```

Comment:
Might be worth either adding a test or modifying this one so that the creation also happens as a part of the Batch.